compiler/src/dotty/tools/dotc/ast/tpd.scala

```diff
@@ -1082,21 +1082,27 @@ object tpd extends Trees.Instance[Type] with TypedTreeInfo {
   /** A key to be used in a context property that tracks enclosing inlined calls */
   private val InlinedCalls = new Property.Key[List[Tree]]
 
-  override def inlineContext(call: Tree)(implicit ctx: Context): Context =
-    ctx.fresh.setProperty(InlinedCalls, call :: enclosingInlineds)
-
-  /** All enclosing calls that are currently inlined, from innermost to outermost.
-   *  EmptyTree calls cancel the next-enclosing non-empty call in the list
-   */
-  def enclosingInlineds(implicit ctx: Context): List[Tree] = {
-    def normalize(ts: List[Tree]): List[Tree] = ts match {
-      case t :: (ts1 @ (t1 :: ts2)) if t.isEmpty => normalize(if (t1.isEmpty) ts1 else ts2)
-      case t :: ts1 => t :: normalize(ts1)
-      case Nil => Nil
+  /** Record an enclosing inlined call.
+    * EmptyTree calls (for parameters) cancel the next-enclosing call in the list instead of being added to it.
+    * We assume parameters are never nested inside parameters.
```

Comment:
WIP further experiments suggest this can't be true, even tho this code works.

Reply:
After normalization, we should not have them, so it should be fine or the normalization was not performed correctly.

Reply:
True after normalization — I thought that was also true before normalization, but that was wrong.